NorthstarDLL/logging/crashhandler.cpp

```diff
@@ -1,377 +1,527 @@
+	case EXCEPTION_ACCESS_VIOLATION:         return "EXCEPTION_ACCESS_VIOLATION";
+	case EXCEPTION_DATATYPE_MISALIGNMENT:    return "EXCEPTION_DATATYPE_MISALIGNMENT";
+	case EXCEPTION_BREAKPOINT:               return "EXCEPTION_BREAKPOINT";
+	case EXCEPTION_SINGLE_STEP:              return "EXCEPTION_SINGLE_STEP";
+	case EXCEPTION_ARRAY_BOUNDS_EXCEEDED:    return "EXCEPTION_ARRAY_BOUNDS_EXCEEDED";
+	case EXCEPTION_FLT_DENORMAL_OPERAND:     return "EXCEPTION_FLT_DENORMAL_OPERAND";
+	case EXCEPTION_FLT_DIVIDE_BY_ZERO:       return "EXCEPTION_FLT_DIVIDE_BY_ZERO";
+	case EXCEPTION_FLT_INEXACT_RESULT:       return "EXCEPTION_FLT_INEXACT_RESULT";
+	case EXCEPTION_FLT_INVALID_OPERATION:    return "EXCEPTION_FLT_INVALID_OPERATION";
+	case EXCEPTION_FLT_OVERFLOW:             return "EXCEPTION_FLT_OVERFLOW";
+	case EXCEPTION_FLT_STACK_CHECK:          return "EXCEPTION_FLT_STACK_CHECK";
+	case EXCEPTION_FLT_UNDERFLOW:            return "EXCEPTION_FLT_UNDERFLOW";
+	case EXCEPTION_INT_DIVIDE_BY_ZERO:       return "EXCEPTION_INT_DIVIDE_BY_ZERO";
+	case EXCEPTION_INT_OVERFLOW:             return "EXCEPTION_INT_OVERFLOW";
+	case EXCEPTION_PRIV_INSTRUCTION:         return "EXCEPTION_PRIV_INSTRUCTION";
+	case EXCEPTION_IN_PAGE_ERROR:            return "EXCEPTION_IN_PAGE_ERROR";
+	case EXCEPTION_ILLEGAL_INSTRUCTION:      return "EXCEPTION_ILLEGAL_INSTRUCTION";
+	case EXCEPTION_NONCONTINUABLE_EXCEPTION: return "EXCEPTION_NONCONTINUABLE_EXCEPTION";
+	case EXCEPTION_STACK_OVERFLOW:           return "EXCEPTION_STACK_OVERFLOW";
+	case EXCEPTION_INVALID_DISPOSITION:      return "EXCEPTION_INVALID_DISPOSITION";
+	case EXCEPTION_GUARD_PAGE:               return "EXCEPTION_GUARD_PAGE";
+	case EXCEPTION_INVALID_HANDLE:           return "EXCEPTION_INVALID_HANDLE";
 	}
-	else
-		spdlog::error("Failed to write minidump file {}!", stream.str());
+	// clang-format on
+	return "UNKNOWN_EXCEPTION";
 }
 
-long GenerateExceptionLog(EXCEPTION_POINTERS* exceptionInfo)
+//-----------------------------------------------------------------------------
+// Purpose: Shows a message box
+//-----------------------------------------------------------------------------
+void CCrashHandler::ShowPopUpMessage()
 {
-	storedException->exceptionRecord = *exceptionInfo->ExceptionRecord;
-	storedException->contextRecord = *exceptionInfo->ContextRecord;
-	const DWORD exceptionCode = exceptionInfo->ExceptionRecord->ExceptionCode;
+	if (m_bHasShownCrashMsg)
+		return;
+
+	m_bHasShownCrashMsg = true;
+
+	if (!IsDedicatedServer())
+	{
+		// Create Crash Message dialog
+		STARTUPINFOA si;
+		PROCESS_INFORMATION pi;
 
-	void* exceptionAddress = exceptionInfo->ExceptionRecord->ExceptionAddress;
+		ZeroMemory(&si, sizeof(si));
+		si.cb = sizeof(si);
+		ZeroMemory(&pi, sizeof(pi));
 
-	storedException->cause = GetExceptionName(*storedException);
+		std::string svCmdLine =
+			fmt::format("bin\\CrashMsg.exe {} {} {} {}", GetNorthstarPrefix(), GetExceptionString(), m_svCrashedModule, m_svCrashedOffset);
```

Comment:
Also how would distrobution of this be handled?

Reply:
> Is there any major benefit to executing a remote process instead of just doing it in Northstar?

MessageBox blocks the thread it is called on, from experience in some cases the game likes to hang for quite a while before terminating.

Separating the message box into a separate process means the game can close independent of the user interacting with the message box.

Reply:
> Also how would distrobution of this be handled?

Preferably with a second PR I have in the works, which unfortunately depends on this being merged

Reply:
Feel free to draft PR that one already in case you haven't yet ^^